src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeArrayIfFull();
```

Comment:
better check if you need to resize here and invoke method resize()
```suggestion
        if (size == array.length) {
            resize();
        }
```